src/python/setup.py

```diff
@@ -22,9 +24,38 @@ def is_pure(self):
 
 
 class InstallEngine(install):
-    """Helper class to hook the python setup.py install path to download client libraries and engine"""
+    """Helper class to hook the python setup.py install path to download
+    client libraries and engine
+    """
+
+    user_options = install.user_options + [
+        ("minimal", None, "control minimal installation"),  # a 'flag' option
+    ]
+
+    def initialize_options(self):
+        install.initialize_options(self)
+        self.minimal = None
 
     def run(self):
+        if self.minimal is not None:
+
+            def do_not_install(require):
+                require = require.strip()
+                for name in NON_MINIMAL_LIST:
+                    if require.startswith(name):
+                        return False
+                return True
+
+            install_requires_minimal = list(
```

Comment:
`install_requires_minimal = [r for r in self.distribution.install_requires
                                                 if not any(r.startswith(nm) for nm in NON_MINIMAL_LIST)]`

Reply:
I think my version is more readable, even though it's kind of verbose.